src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,128 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
 
+    private static final int DEFAULT_SIZE = 10;
+    private Object[] elementsData;
+    private Object[] emptyArray = {};
+    private int size;
+
+    public ArrayList() {
+        elementsData = emptyArray;
+        size = 0;
+    }
+
+    public ArrayList(int arraySize) {
+        size = 0;
+        if (arraySize > 0) {
+            elementsData = new Object[arraySize];
+        } else if (arraySize == 0) {
+            elementsData = emptyArray;
+        } else {
+            throw new IllegalArgumentException("Incorrect size of array");
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elementsData.length) {
+            elementsData = grow();
+        }
+        elementsData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndex(index);
+        if (size == elementsData.length) {
+            elementsData = grow();
+        }
+        System.arraycopy(elementsData, index, elementsData, index + 1, size - index);
+        elementsData[index] = value;
+        size += 1;
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list == null) {
+            return;
+        }
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
 
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return (T) elementsData[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index);
+        elementsData[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        int numMoved = size - index - 1;
+        T oldValue = (T) elementsData[index];
+        System.arraycopy(elementsData, index + 1, elementsData, index, numMoved);
+        --size;
+        return oldValue;
     }
 
     @Override
     public T remove(T t) {
-        return null;
+        int i = 0;
+        if (t == null) {
+            for (; i < size; i++) {
```

Comment:
please, simplify. one loop is just enough :)

Reply:
ok